src/appPackageManifest.ts

```diff
@@ -77,6 +84,25 @@ class AppPackageManifestTransform extends Transform {
     super({ objectMode: true });
   }
 
+  private populateTileData(): void {
+    if (this.projectConfig.appType === AppType.APP) {
+      const appConfig = this.projectConfig as AppProjectConfiguration;
+
+      if (appConfig.tiles !== undefined) {
+        let componentTiles: GalleryTile[] = [];
+        for (let i = 0; i < appConfig.tiles.length; i++) {
```

Comment:
You can simplify this to `for (const tile of appConfig.tiles) {`

or simpler still:

```
const componentTiles = appConfig.tiles.map(tile => ({
  name: tile:name,
  id: tile.uuid,
  platforms: tile.buildTargets !== undefined ? tile.buildTargets as string[] : appConfig.buildTargets,
});
```

Reply:
Updated approach. Please take another look.